src/refract/JSONSchemaVisitor.h

```diff
@@ -36,8 +37,28 @@ namespace refract
         void enumElement(const ArrayElement& e, const ArrayElement::ValueType *val);
         bool allItemsEmpty(const ArrayElement::ValueType* val);
 
+        template<typename T>
+        void setPrimitiveType(const T& e)
+        {
+            static_assert(sizeof(T) == 0, "Only String, Number, Boolean types allowed");
+        };
+
+        template<typename T>
+        void primitiveType(const T& e, const typename T::ValueType *v)
```

Comment:
Not sure, but I mean body of private template member function can be just "declared" in header and definition can be moved in to `cc`. we need no to expose it.